pkgs/by-name/ch/chatall/package.nix

```
@@ -0,0 +1,94 @@
+{ lib
+, fetchFromGitHub
+, buildNpmPackage
+, makeDesktopItem
+, nix-update-script
+
+, copyDesktopItems
+, electron
+, icoutils
+, jq
+, makeWrapper
+, moreutils
+}:
+
+buildNpmPackage rec {
+  pname = "chatall";
+  version = "1.51.79";
+
+  src = fetchFromGitHub {
+    owner = "sunner";
+    repo = "ChatAll";
+    rev = "v${version}";
+    hash = "sha256-Gi08TdD6Wl7PUPmNJdjtaMsEdZB2CU1X2I+dhPv1O5U=";
+  };
+
+  npmDepsHash = "sha256-pDuIoAsUEhwJ7TLpslE7VRhoXD3f0X5ipVjtrJmGwPI=";
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    icoutils
+    makeWrapper
+  ];
+
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = true;
+  };
+
+  makeCacheWritable = true;
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "ChatALL";
+      exec = "chatall";
+      icon = "chatall";
+      comment = meta.description;
+      categories = [ "Utility" "Chat" ];
+    })
+  ];
+
+  postPatch = ''
+    ${jq}/bin/jq \
+      '.main = "dist_electron/bundled/index.js" |
+      del(.dependencies."electron-builder")' \
+      package.json | ${moreutils}/bin/sponge package.json
```

```suggestion
      package.json | sponge package.json
```
Using ${package}/bin/... will not work when cross-compiling.